zenith_utils/build.rs

```diff
@@ -0,0 +1,8 @@
+use std::env;
+
+fn main() {
+    let version = env::var("GIT_VERSION").unwrap_or_else(|_| "unknown".to_string());
+
+    println!("cargo:rustc-env=GIT_VERSION=git-env:{}", version);
+    println!("cargo:rerun-if-env-changed=GIT_VERSION");
```

Comment:
Yes, or if the build script does it. Not that important, just that it's confusing to set `GIT_VERSION=bee1` in the script and then read it as `git-env:bee1` in Rust.

Reply:
Didnt get the second part. You mean that rebuild should be triggered if any environment variable has changed? This would result in spurious rebuilds. And AFAIK just rerun-if-env-changed without parameters is not supported.

Remember having bad experience with [bazel](https://bazel.build/) because it randomly invalidated the cache when env/execution dir has changed. (it was doing so on purpose but it was so annoying)

Reply:
I meant just the `rerun-if-env-changed=GIT_VERSION` without actually setting `rustc-env`. I believe the former is needed so that the crate is rebuilt on env changes, and the latter is only for prepending `git-env:` which can be done in `lib.rs`

Reply:
>  just that it's confusing 

IMO thats OK because the prefix is `git-env:` so it's kind of descriptive to me that this is the git version that came from env. If it was named just `VERSION`, then that indeed would be confusing.

Reply:
Yes, prepending `git-env:` is absolutely ok with me. Prepending it in the guts of the build system, IMHO, is less clear than prepending it either where the variable is initially set or where it finally makes its way into Rust code.

Reply:
Yeah, got it, makes sense.